pkg/controller/jobframework/setup.go

```diff
@@ -99,6 +105,40 @@ func (m *integrationManager) setupControllers(mgr ctrl.Manager, log logr.Logger,
 	})
 }
 
+func (m *integrationManager) setupControllerAndWebhook(mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
+	if err := cb.NewReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, options.ManagerName)),
+		opts...,
+	).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("%s: %w", fwkNamePrefix, err)
+	}
+	if err := cb.SetupWebhook(mgr, opts...); err != nil {
+		return fmt.Errorf("%s: unable to create webhook: %w", fwkNamePrefix, err)
+	}
+	m.enableIntegration(name)
+	return nil
+}
+
+func (m *integrationManager) waitForAPI(ctx context.Context, mgr ctrl.Manager, log logr.Logger, gvk schema.GroupVersionKind, action func()) {
+	for {
+		m.mu.Lock()
```

Comment:
I've been finding this lock hard to avoid as a `write` is performed in the tests [[here] ](https://github.com/kubernetes-sigs/kueue/pull/2574/files#diff-0e0e2357162d6c7d6172e1f1e2f0fed3d27946005ee221718652d972f6a61879R167)

However, I enhanced this by using `RLock()` instead which can be held by an arbitrary number of readers.

Reply:
Actually, I don't like that we are using the same mutex for `mgr.GetRESTMapper()` and `integrationManager`. It's not related for integrationManager and need just for test. But we will unnecessary lock this code on update `integrationManager`. In my opinion it's better to create another one just for test.

Reply:
That makes sense, I've added a new mutex for use with the RESTMapper. Let me know what you think, thank you.